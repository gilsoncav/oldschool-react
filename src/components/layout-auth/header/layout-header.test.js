import LayoutHeader from './layout-header';

describe('LayoutHeader component', () => {
  it('renders correctly', () => {
    const spy = sinon.spy();
    const historyMock = { push: spy };
    const wrapper = shallow(<LayoutHeader.WrappedComponent history={historyMock} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should setState of loadingLogout on logout-btn Click', async (done) => {
    const spy = sinon.spy();
    const historyMock = { push: spy };
    const wrapper = shallow(<LayoutHeader.WrappedComponent history={historyMock} />);

    wrapper.setState({ loadingLogout: false });

    await wrapper.instance().logout();
    expect(spy.calledOnce).toBe(true);
    expect(spy.args[0][0]).toBe('/');
    expect(wrapper.state().loadingLogout).toBe(false);
    done();
  });
});
