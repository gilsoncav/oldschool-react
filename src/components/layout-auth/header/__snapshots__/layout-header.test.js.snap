// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutHeader component renders correctly 1`] = `
<Adapter
  style={
    Object {
      "paddingLeft": "5px",
      "paddingRight": "5px",
      "textAlign": "center",
    }
  }
>
  <div
    className="logo"
    style={
      Object {
        "float": "left",
      }
    }
  >
    <Button
      block={false}
      className="btn-secondary"
      ghost={false}
      htmlType="button"
      icon="menu-fold"
      loading={false}
      size="large"
    />
    <Link
      to="/dashboard"
    >
      <img
        alt="Logo"
        src="logo.png"
        style={
          Object {
            "marginLeft": "15px",
            "marginRight": "30px",
            "width": "120px",
          }
        }
      />
    </Link>
  </div>
  <Input
    disabled={false}
    id="header-search"
    placeholder="Search Games"
    prefix={
      <Icon
        style={
          Object {
            "color": "#b3b3b3",
          }
        }
        type="search"
      />
    }
    size="large"
    style={
      Object {
        "width": "50%",
      }
    }
    type="text"
  />
  <div
    className="logo"
    style={
      Object {
        "float": "right",
      }
    }
  >
    <Button
      block={false}
      className="btn-secondary"
      ghost={false}
      htmlType="button"
      icon="logout"
      id="logout-btn"
      loading={false}
      onClick={[Function]}
      size="large"
    />
  </div>
</Adapter>
`;
